Composer/packages/lib/indexers/src/dialogUtils/jsonDiff.ts

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import has from 'lodash/has';
+import get from 'lodash/get';
+import set from 'lodash/set';
+import isEqual from 'lodash/isEqual';
+import cloneDeep from 'lodash/cloneDeep';
+
+import { JsonWalk, VisitorFunc } from '../utils/jsonWalk';
+
+import { ListDiff } from './listDiff';
+
+export interface IJSONChangeAdd {
+  path: string;
+  value: any;
+}
+export interface IJSONChangeDelete {
+  path: string;
+  value: any;
+}
+export interface IJSONChangeUpdate {
+  path: string;
+  value: any;
+  preValue: any;
+}
+
+export interface IJsonChanges {
+  adds: IJSONChangeAdd[];
+  deletes: IJSONChangeDelete[];
+  updates: IJSONChangeUpdate[];
+}
+
+export type IComparator = (json1: any, json2: any, path: string) => { isChange: boolean; isStop: boolean };
```

Comment:
In this comparator interface, too much concept here, there is a path, there is IsChange and isStop and no result returned, which is weird. 

what a `comparator` should represent, in almost any programming language, it usually takes two object of same type, and result in a true\false comparasion result.